### ISSUE TITLE
IDISP0013 should not warn when not using the disposable

### ISSUE THREAD

**Issue by JohanLarsson** [DotNetAnalyzers/IDisposableAnalyzers#76]
```cs
namespace RoslynSandbox
{
    using System.IO;
    using System.Threading.Tasks;

    public class Foo
    {
        public static Task FooAsync()
        {
            using (var stream = File.OpenRead(string.Empty))
            {
                return Task.Delay(0);
            }
        }
    }
}
```

**Comment by JohanLarsson** [comment]
Not sure how to write this rule as it can be:
```cs
using (Lock())
{
    return Task.Delay(0);
}
```

Where we probably want to await.

**Comment by JeremyMorton** [comment]
I am seeing this with `ValueTask<>` return types:

![image](https://user-images.githubusercontent.com/10471528/230499094-d3af1aee-0a81-4fec-8914-94180b800bec.png)

```
  public ValueTask<int> M1Async()
  {
    using (var disposable = new Disposable())
    {
      return new ValueTask<IDisposable>(disposable.Equals(disposable) ? 1 : 0);
    }
  }

  public ValueTask<int> M2Async()
  {
    using (var disposable = new Disposable())
    {
      var i = disposable.Equals(disposable) ? 1 : 0;
      return new ValueTask<IDisposable>(i);
    }
  }
```

**Comment by JohanLarsson** [comment]
@JeremyMorton it is easier for me if you create new issues when you find bugs, never hesitate to create issues.

**Comment by JeremyMorton** [comment]
Thanks, I wasn't sure where searching for existing issues vs them already being closed balanced out, but not reviving closed ones makes sense.
